include/autoppl/expression/binop.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <type_traits>
+#include <autoppl/util/traits.hpp>
+#include <autoppl/expression/var_expr.hpp>
+#include <autoppl/expression/variable.hpp>
+
+namespace ppl {
+
+template <class LHSValueType, class RHSValueType, class BinaryOp>
+struct BinaryOpNode 
+{
```

Comment:
can now add
```cpp
static_assert(util::is_var_expr_v<LHSVarExprType>);
static_assert(util::is_var_expr_v<RHSVarExprType>);
```
You may have to `#include <autoppl/util/var_expr_traits.hpp>`. This is how we constrain these types to fit the concept of "variable expression".